### ISSUE TITLE
Feature Request PrintWidth

### ISSUE THREAD

**Issue by Mister-Hope** [hilleer/vscode-yaml-plus-json#19]
![image](https://user-images.githubusercontent.com/33315834/86881175-72f51500-c120-11ea-98f7-dfc81e77892e.png)
I noticed that this plugin is cutting long string for us, but in my cases, the printWidth is a bit large.

Could you add a `printWidth` config item to let users config it? Thanks! 

And just a reminder, you should minus the spaces at the start of the line. Don't forget that

**Comment by Mister-Hope** [comment]
Besides, currently, the lines are not aligned either.
![image](https://user-images.githubusercontent.com/33315834/86881490-0a5a6800-c121-11ea-9684-46063d24081e.png)

**Comment by hilleer** [comment]
Hi @Mister-Hope ,

Thanks for your request.

Can you by any chance share these files that you tried to convert? :-)

**Comment by hilleer** [comment]
@Mister-Hope I fail to reproduce it. Is it possible for you to share the original file you converted from or create a similar one where the same issue occurs?

**Comment by Mister-Hope** [comment]
> @Mister-Hope I fail to reproduce it. Is it possible for you to share the original file you converted from or create a similar one where the same issue occurs?

Sure. but It's 2 am now, I just saw it before going to bed. I will share some files tomorrow.

**Comment by Mister-Hope** [comment]
```json
{
  "title": "学生心理健康教育中心",
  "desc": "信息来源: 学生处官网",
  "time": "2020年9月1日",
  "shareable": true,
  "content": [
    {
      "tag": "title",
      "text": "学生心理健康教育中心简介"
    },
    {
      "tag": "text",
      "type": "p",
      "text": [
        "   东北师范大学学生心理健康教育中心是面向全校学生实施心理健康教育和开展心理咨询服务的专业机构。我校于1989年3月成立大学生心理与行为指导中心，2004年在此基础上成立学生心理健康教育中心，隶属学生处，科级建制。2013年8月，学校将学生心理健康教育中心由科级建制，发展为副处级建制，挂靠学生处。中心核定人员编制8人，设主任岗位1个，下设教育培训部、测验咨询部、心理危机干预部三个内设机构。学校成立由校长任组长、主管学生工作和研究生工作的校领导任副组长，学生处、教务处等8个相关部门领导为成员的心理健康教育督导组，整体规划和督导大学生心理健康教育工作的实施。中心的工作理念是以全体学生的健康成长和全面发展为中心，教育为本，发展为主，预防为辅，提高学生的心理素质，更好地适应大学生活与未来社会。",
        "   根据教育部办公厅下发的关于进一步加强高校心理健康教育工作的相关文件精神及当前的形势需要，学校以“提供国内一流的专业化服务”为目标，构建心理健康教育工作体系，组建高质量的心理咨询师队伍，夯实中心快速发展的软硬环境基础。在队伍建设方面，我校构建“四有”心理健康教育队伍，在中心层面，聘请心理学、教育学、思想政治教育、心理卫生等领域的专家15人，具体指导心理咨询、心理危机干预与心理培训工作；在学院层面，每个学院辅导员中设有“心理专员”，负责各学院心理健康教育工作的落实；在学生层面，遴选出心理素质好的学生党员、学生干部为“心理委员”，通过朋辈辅导和互助，引导学生实现自我心理成长。学校通过“四位一体”的心理健康教育工作队伍，确保了学生心理危机排查与处理工作迅速、科学、有效地开展，确保了学校心理健康教育工作有力地组织与实施。在软硬件建设方面，中心设有3间面谈咨询室、4间团体咨询室、1间心理测量室、10间谈心辅导室。同时，学校投资近10万元购买了先进的心理咨询辅助设备，如肌电生物反馈仪，心理测评系统，为心理健康教育与咨询工作的专业化、规范化提供了有力保障。",
        "   我校探索实施四个“四位一体”的心理健康教育工作模式，构建“四有”心理健康教育队伍，在教育部组织心理学专家召开的心理健康教育研讨会上，我校就心理健康教育工作模式作经验汇报，是四所汇报学校之一；持续开展“优化素质·幸福成长”学生心理素质培养工程，不断完善“预防教育—鉴别排查—危机干预—跟踪监控”四位一体的心理危机干预体系。《教育部简报》以《东北师范大学切实抓好学生心理健康动态监控和咨询服务》和《东北师范大学积极建设心理健康教育与咨询“四有”队伍》为题对我校心理健康教育工作予以专门报道。"
      ]
    },
    {
      "tag": "title",
      "text": "心理咨询服务安排"
    },
    {
      "tag": "text",
      "type": "p",
      "text": "   为了更好地帮助学生解决其在自我发展、人际交往、学习生活、情绪调节、生涯规划、求职择业等方面存在的心理困惑、心理问题，学校将为学生提供多渠道的心理咨询服务，具体安排如下:"
    },
    {
      "tag": "text",
      "type": "ul",
      "heading": "面谈咨询",
      "text": [
        "预约时间: 每天13:30—17:00；17:30—20:30",
        "预约电话: 0431-85099129 或 9129(内线，本部净月一样)",
        "面谈时间: 每周一至周五13:30—17:00",
        "面谈地点:\n本部校区:\n学生活动中心(北苑食堂)三楼右侧A07室\n净月校区:\n学生活动中心二楼(一食堂旁)202室"
      ]
    },
    {
      "tag": "text",
      "heading": "热线咨询",
      "text": "服务时间: 每天13:30—17:00；17:30—20:30\n热线电话: 0431-85099129 或 9129(内线)"
    },
    {
      "tag": "text",
      "heading": "邮箱咨询",
      "text": "电子邮箱: xinyun@nenu.edu.cn，会在 3 日内回复"
    },
    {
      "tag": "text",
      "heading": "在线咨询",
      "text": "咨询时间: 每天13:30—17:00；17:30—20:30\nQQ号: 1002759092"
    },
    {
      "tag": "text",
      "type": "p",
      "heading": "朋辈辅导员的帮助",
      "text": "“学生朋辈辅导员”是学生处学生心理健康教育中心指导下的学生团体组织，是学生身边的贴心朋友。他们需要掌握心理科学基础知识，运用心理辅导技能，帮助同学处理好学习成才、择业交友、健康生活等方面的具体问题，同时也需要策划、组织学院心理健康教育活动。"
    },
    {
      "tag": "text",
      "heading": "其他咨询服务方式",
      "text": "生涯咨询预约电话: 0431-85098816"
    }
  ]
}
```


will be convert to 

```yaml
title: 学生心理健康教育中心
desc: "信息来源: 学生处官网"
time: 2020年9月1日
shareable: true
content:
  - tag: title
    text: 学生心理健康教育中心简介
  - tag: text
    type: p
    text:
      - "   东北师范大学学生心理健康教育中心是面向全校学生实施心理健康教育和开展心理咨询服务的专业机构。我校于1989年3月成立大学生心理与行为指\
        导中心，2004年在此基础上成立学生心理健康教育中心，隶属学生处，科级建制。2013年8月，学校将学生心理健康教育中心由科级建制，发展为副处级\
        建制，挂靠学生处。中心核定人员编制8人，设主任岗位1个，下设教育培训部、测验咨询部、心理危机干预部三个内设机构。学校成立由校长任组长、主管学生\
        工作和研究生工作的校领导任副组长，学生处、教务处等8个相关部门领导为成员的心理健康教育督导组，整体规划和督导大学生心理健康教育工作的实施。中心\
        的工作理念是以全体学生的健康成长和全面发展为中心，教育为本，发展为主，预防为辅，提高学生的心理素质，更好地适应大学生活与未来社会。"
      - "   根据教育部办公厅下发的关于进一步加强高校心理健康教育工作的相关文件精神及当前的形势需要，学校以“提供国内一流的专业化服务”为目标，构建\
        心理健康教育工作体系，组建高质量的心理咨询师队伍，夯实中心快速发展的软硬环境基础。在队伍建设方面，我校构建“四有”心理健康教育队伍，在中心层面\
        ，聘请心理学、教育学、思想政治教育、心理卫生等领域的专家15人，具体指导心理咨询、心理危机干预与心理培训工作；在学院层面，每个学院辅导员中设有\
        “心理专员”，负责各学院心理健康教育工作的落实；在学生层面，遴选出心理素质好的学生党员、学生干部为“心理委员”，通过朋辈辅导和互助，引导学生实\
        现自我心理成长。学校通过“四位一体”的心理健康教育工作队伍，确保了学生心理危机排查与处理工作迅速、科学、有效地开展，确保了学校心理健康教育工作\
        有力地组织与实施。在软硬件建设方面，中心设有3间面谈咨询室、4间团体咨询室、1间心理测量室、10间谈心辅导室。同时，学校投资近10万元购买了先\
        进的心理咨询辅助设备，如肌电生物反馈仪，心理测评系统，为心理健康教育与咨询工作的专业化、规范化提供了有力保障。"
      - "   我校探索实施四个“四位一体”的心理健康教育工作模式，构建“四有”心理健康教育队伍，在教育部组织心理学专家召开的心理健康教育研讨会上，我\
        校就心理健康教育工作模式作经验汇报，是四所汇报学校之一；持续开展“优化素质·幸福成长”学生心理素质培养工程，不断完善“预防教育—鉴别排查—危机\
        干预—跟踪监控”四位一体的心理危机干预体系。《教育部简报》以《东北师范大学切实抓好学生心理健康动态监控和咨询服务》和《东北师范大学积极建设心理\
        健康教育与咨询“四有”队伍》为题对我校心理健康教育工作予以专门报道。"
  - tag: title
    text: 心理咨询服务安排
  - tag: text
    type: p
    text: "   为了更好地帮助学生解决其在自我发展、人际交往、学习生活、情绪调节、生涯规划、求职择业等方面存在的心理困惑、心理问题，学校将为学生提供多渠道\
      的心理咨询服务，具体安排如下:"
  - tag: text
    type: ul
    heading: 面谈咨询
    text:
      - "预约时间: 每天13:30—17:00；17:30—20:30"
      - "预约电话: 0431-85099129 或 9129(内线，本部净月一样)"
      - "面谈时间: 每周一至周五13:30—17:00"
      - |-
        面谈地点:
        本部校区:
        学生活动中心(北苑食堂)三楼右侧A07室
        净月校区:
        学生活动中心二楼(一食堂旁)202室
  - tag: text
    heading: 热线咨询
    text: |-
      服务时间: 每天13:30—17:00；17:30—20:30
      热线电话: 0431-85099129 或 9129(内线)
  - tag: text
    heading: 邮箱咨询
    text: "电子邮箱: xinyun@nenu.edu.cn，会在 3 日内回复"
  - tag: text
    heading: 在线咨询
    text: |-
      咨询时间: 每天13:30—17:00；17:30—20:30
      QQ号: 1002759092
  - tag: text
    type: p
    heading: 朋辈辅导员的帮助
    text: “学生朋辈辅导员”是学生处学生心理健康教育中心指导下的学生团体组织，是学生身边的贴心朋友。他们需要掌握心理科学基础知识，运用心理辅导技能，帮助同学处理好学习成才、择业交友、健康生活等方面的具体问题，同时也需要策划、组织学院心理健康教育活动。
  - tag: text
    heading: 其他咨询服务方式
    text: "生涯咨询预约电话: 0431-85098816"
```

**Comment by Mister-Hope** [comment]
The string cut of by `\` is beginning with three spaces.

If I remove them, they won't be cut off

**Comment by Mister-Hope** [comment]
Also a string ends with a `:` will be cut off.

```json
{
  "title": "学生心理健康教育中心",
  "desc": "信息来源: 学生处官网",
  "time": "2020年9月1日",
  "shareable": true,
  "content": [
    {
      "tag": "text",
      "type": "p",
      "text": "啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊:"
    },
    {
      "tag": "text",
      "type": "p",
      "text": "啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊"
    }
  ]
}
```

will be convert to

```yaml
title: 学生心理健康教育中心
desc: "信息来源: 学生处官网"
time: 2020年9月1日
shareable: true
content:
  - tag: text
    type: p
    text: "啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊\
      啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊\
      啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊:"
  - tag: text
    type: p
    text: 啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊
```

**Comment by Mister-Hope** [comment]
Any progress on this feature?

**Comment by hilleer** [comment]
Hi @Mister-Hope,

I have not had time to look into it yet, no. I will look into it asap.

Meanwhile, feel free to open a pull-request to speed up the progress :-)

**Comment by hilleer** [comment]
Hi @Mister-Hope ,

I am sorry this had to take so long. Based on my findings, this appears to be due to the the behaviour of the module used to do the conversion. It has an option `schema` that defaults to `core`. Instead of defaulting to `core` but instead use`json` it appears to resolve this "issue". See more [here](https://github.com/eemeli/yaml/blob/master/docs/03_options.md#data-schemas).

I have published version `1.6.0` that adds a configuration `yamlSchema` you can use to configure it to use schema `json`.
